pv211_utils/beir/eval.py

```diff
@@ -0,0 +1,34 @@
+from typing import Set, Optional
+
+from pv211_utils.eval import EvaluationBase
+from .entities import BeirJudgementBase
+from .irsystem import BeirIRSystemBase
+
+
+class BeirEvaluation(EvaluationBase):
+    """A Generic information retrieval system evaluation.
+
+    Parameters
+    ----------
+    system : BeirIRSystemBase
+        The information retrieval system.
+    judgements : set of BeirJudgementBase
+        Pairs of queries and relevant documents.
+    k : int, optional
+        Parameter defining evaluation depth. Default is 10.
+    leaderboard : BeirLeaderboard or None, optional
+        A leaderboard to which we may later submit evaluation results.
+        If None, then evaluation results will not be submitted. Default is None.
+    author_name : str or None, optional
+        The name of the author submitted to the leaderboard.
+        If None, then the result will not be submitted. Default is None.
+    num_workers : int or None, optional
+        The number of processes used to compute the mean average precision.
+        If None, all available CPUs will be used. Default is 1.
+    """
+    def __init__(self, system: BeirIRSystemBase, judgements: Set[BeirJudgementBase], k: Optional[int] = 10,
+                 leaderboard=None, author_name: Optional[str] = None, num_workers: Optional[int] = 1):
+        super().__init__(system, judgements, k, leaderboard, author_name, num_workers)
+
+    def _get_minimum_mean_average_precision(self) -> float:
+        return 0.1
```

Comment:
@MarekToma Please, conduct an experiment [similar to the one we did for TREC last spring][1]. As discussed on Monday, skip TF-IDF and focus just on BM25 using the implementation from `systems/bm25.py`. Report a table with the results [to GitLab][2], so that we can discuss what the minimum MAP score should be (both here and in the first markdown paragraph of `notebooks/beir_cqadupstack.ipynb`).

 [1]: https://gitlab.fi.muni.cz/sojka/mir-general/-/issues/80#note_210734
 [2]: https://gitlab.fi.muni.cz/sojka/mir-general/-/issues/90

Reply:
working on it

Reply:
Seems done: https://gitlab.fi.muni.cz/sojka/mir-general/-/issues/90#note_358749

Reply:
@MarekToma After we have decided on the minimum MAP score on GitLab, please:

- [ ] Add the minimum required MAP score to the assignment at the top of the notebook.
- [ ] Add an example solution to the notebook that will achieve less than the minimum required MAP score.
- [ ] Link the notebook from `README.md`.